src/main/java/chess/domain/board/ChessBoard.java

```diff
@@ -0,0 +1,94 @@
+package chess.domain.board;
+
+import static chess.domain.board.InitialPieces.INITIAL_PIECES;
+
+import chess.domain.Position;
+import chess.domain.piece.Piece;
+import chess.domain.piece.PieceMoveResult;
+import chess.domain.piece.Team;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class ChessBoard {
+    private final List<Piece> piecesOnBoard;
+    private Team currentTeam = Team.WHITE;
+
+    public ChessBoard() {
+        this(INITIAL_PIECES);
+    }
+
+    public ChessBoard(List<Piece> pieces) {
+        piecesOnBoard = new ArrayList<>(pieces);
+    }
+
+    public ChessBoard(Piece... pieces) {
+        this(List.of(pieces));
+    }
+
+    boolean move(Position from, Position to) {
+        if (isEmptyPosition(from) || isOtherTeamTurn(from)) {
+            return false;
+        }
+        Piece piece = findPiece(from);
+        PieceMoveResult moveResult = piece.move(to, this);
+        removePieceIfCaught(to, moveResult);
+        changeCurrentTeamIfNotFail(moveResult);
+        return moveResult.toBoolean();
+    }
+
+    private boolean isEmptyPosition(Position from) {
+        Optional<Piece> optionalPiece = piecesOnBoard.stream()
+                .filter(piece1 -> piece1.isOn(from))
+                .findFirst();
+        return optionalPiece.isEmpty();
+    }
```

Comment:
```suggestion
    private boolean isEmptyPosition(Position from) {
        return piecesOnBoard.stream()
                .noneMatch(piece -> piece.isOn(from));
    }
```

위와 같이 수정하는 건 어떨까요? 테스트는 모두 통과하네요!